src/Entity.ts

```diff
@@ -16,10 +16,10 @@ export class Entity {
   }
 
   protected _query<T>(
-    keys: (string | number)[] | null,
+    keys: (string | number | undefined)[] | null,
     observableCallback: () => Observable<T>,
     options?: CentrifugeQueryOptions
   ) {
-    return this._root._query<T>(keys ? [...this.#baseKeys, ...keys] : null, observableCallback, options)
+    return this._root._query<T>(keys ? [...this.#baseKeys, ...keys.filter(Boolean)] : null, observableCallback, options)
```

Comment:
I think we shouldn't filter here, because `undefined` can be stringified just fine and it may lead to an edge case where two distinct keys then share the same cache key. e.g.:
`[type, filter?.from, filter?.to, filter?.groupBy]` might have keys `['balaceSheet', '01-01-2024', undefined, 'month']`, or `['balaceSheet', undefined, '01-01-2024', 'month']`. If `undefined` gets filtered out we'd have an unwanted overlap of those two keys